### PR TITLE
Update Verify Zend Framework composer dependencies section for Cloud Update Magento Version doc

### PR DESCRIPTION
**Pull request by amcguireweb**
## Purpose of this pull request

Correction the instructions for checking and updating the `auto-load: "psr-4"` when migrating from Magento v2.2 to 2.3 or later.

## Affected DevDocs pages

https://devdocs.magento.com/cloud/project/project-upgrade.html#verify-zend-framework-composer-dependencies

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@amcguireweb thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by dobooth** [comment]
Hi @amcguireweb Thanks for the contribution. Please fill out the description and sign the Adobe CLA, so we can process this PR. Thanks!

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Please fill out the PR description properly. I had to check some Cloud composer.json files.
This information can also be used to search on in the future.

File: src/cloud/project/project-upgrade.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ To check and update Zend Framework dependencies:
          "Magento\\Framework\\": "lib/internal/Magento/Framework/",
          "Magento\\Setup\\": "setup/src/Magento/Setup/",
          "Magento\\": "app/code/Magento/",
+         "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
    +     "Laminas\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"
```
Comment (meker12) [inline src/cloud/project/project-upgrade.md:125]: so I am not sure if the updated version should have both lines, or only the Laminas line.

"Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
"Laminas\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"

@oshmyheliuk  Can you confirm?

In `composer.json` for 2.4 metapackage, only the first line is included: 

https://github.com/magento/magento-cloud/blob/879c6b42bfcb134536364563c9e83f5d74dc3309/composer.json#L33

**Comment by amcguireweb** [comment]
I was told by Magento Commerce aCloud support that it should be the following:

```
 "autoload": {
        "psr-4": {
            "Magento\\Framework\\": "lib/internal/Magento/Framework/",
            "Magento\\Setup\\": "setup/src/Magento/Setup/",
            "Magento\\": "app/code/Magento/",
            "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
            "Laminas\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"
        },
```
This is the only way I could get `composer update` to work (other than not updating at all).

If not this way, how do we migrate Zend to Laminas if not here?

**Comment by meker12** [comment]
> I was told by Magento Commerce Cloud support that it should be the following:
> 
> ```
>  "autoload": {
>         "psr-4": {
>             "Magento\\Framework\\": "lib/internal/Magento/Framework/",
>             "Magento\\Setup\\": "setup/src/Magento/Setup/",
>             "Magento\\": "app/code/Magento/",
>             "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
>             "Laminas\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"
>         },
> ```
> 
> This is the only way I could get `composer update` to work (other than not updating at all).
> 
> If not this way, how do we migrate Zend to Laminas if not here?

-------------------
The DevDocs instructions are incorrect, and this has caused confusion. First, the update to `composer.json` is required only if you are upgrading from 2.2.x version to 2.3.x or later.

If you check the [Magento DevBlog post](https://community.magento.com/t5/Magento-DevBlog/Migration-of-Zend-Framework-to-the-Laminas-Project/ba-p/443251),  there's information about why the root composer.json file doesn't change. 

```We continue to use Zend namespace for \Zend\Mvc\Controller\LazyControllerAbstractFactory to avoid any changes to the autoload section of the root composer.json file. ```

(Another article that mentions this fix for reference: https://mage2.pro/t/topic/5967)


Based on response from Cloud development, the correct section is:

```
 "autoload": {
        "psr-4": {
            "Magento\\Framework\\": "lib/internal/Magento/Framework/",
            "Magento\\Setup\\": "setup/src/Magento/Setup/",
            "Magento\\": "app/code/Magento/",
            "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
        },
```
If you look at the template for Magento v2.4, you'll see that this line is already present.  

```"Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",

If you are upgrading from a version earlier than v2.3, then you needed to add this line. If you are upgrading from 2.3 to 2.4, then the dependency is already present. And you don't need to add it.

**Comment by meker12** [comment]
Proposed changes to clarify instructions:


### Verify Zend composer dependencies

When upgrading to **2.3.x or later from 2.2.x**, verify that the Zend Framework dependencies has been added to the `autoload` property of the `composer.json` file to support Laminas. See [Migration of Zend Framework to the Laminas Project](https://community.magento.com/t5/Magento-DevBlog/Migration-of-Zend-Framework-to-the-Laminas-Project/ba-p/443251) on the _Magento DevBlog_.

{:.procedure}
To check and update Zend Framework dependencies:

1. On your local workstation, change to the Cloud project root directory.

1. Checkout your integration branch.

1. Open the `composer.json` file in a text editor.

1. Check the `autoload:psr-4` section for the Zend plugin manager implementation for controllers dependency:

   ```conf
    "autoload": {
      "psr-4": {
         "Magento\\Framework\\": "lib/internal/Magento/Framework/",
         "Magento\\Setup\\": "setup/src/Magento/Setup/",
         "Magento\\": "app/code/Magento/",
         "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/",
      },
   ```
   {:.no-copy}

1. If the Zend dependency is missing, update `composer.json`:

   -  Add the following line to the `autoload:psr-4` section.

      ```json
      "Zend\\Mvc\\Controller\\": "setup/src/Zend/Mvc/Controller/"
      ```

   -  Update the project dependencies.

      ```bash
      composer update
      ```

   -  Add, commit, and push code changes.

      ```bash
      git add -A
      ```

      ```bash
      git commit -m "Add Zend plugin manager implementation for controllers dependency for Laminas support"
      ```

**Comment by amcguireweb** [comment]
> @amcguireweb Do you mind updating your PR? Or I can create a new one from your PR to make the changes.

Thanks @meker12, I think I updated it. :)

**Review by meker12 — APPROVED** [review]
Thanks for correcting this information @amcguireweb.

**Review by BarnyShergold — APPROVED** [review]

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**